plugins/schedule/base/lib/kScheduleEventsConsumer.php

```diff
@@ -120,19 +172,60 @@ public function updateScheduleEvent($eventId)
         }
     }
 
-    protected function scheduleEventChanged(ScheduleEvent $scheduleEvent, $modifiedColumns)
-    {
-        if (in_array($scheduleEvent->getStatus(), array(ScheduleEventStatus::DELETED, ScheduleEventStatus::CANCELLED)))
-        {
-            $scheduleEvents = ScheduleEventResourcePeer::retrieveByEventId($scheduleEvent->getId());
-            foreach ($scheduleEvents as $currScheduleEvent)
-            {
-                /**
-                 * @var ScheduleEventResource $currScheduleEvent
-                 */
-                $currScheduleEvent->delete();
-            }
-        }
-    }
+	protected function scheduleEventChanged(ScheduleEvent $scheduleEvent, $modifiedColumns = null)
+	{
+		if (in_array($scheduleEvent->getStatus(), array(ScheduleEventStatus::DELETED, ScheduleEventStatus::CANCELLED)))

```

Comment:
Lets break entire function to sub function - 
scheduleEventChangedEventDeleted
scheduleEventChangedEndTimeChanged
scheduleEventChangedLinkedToChanged

Reply:
done